### DISCUSSION TITLE
CPU temperature for non RPI devices

### DISCUSSION THREAD

**Discussion by Vendo233** [hjelev/rpi-mqtt-monitor#152] [General]
I`m using this great code for Proxmox to report sensor data to Home Assistant. The CPU temp is broken in last version

here is what works for me.
`def check_cpu_temp():
    full_cmd = "cat /sys/class/thermal/thermal_zone*/temp 2> /dev/null | sed 's/\\(.\\)..$//' | tail -n 1"
    try:
        p = subprocess.Popen(full_cmd, shell=True, stdout=subprocess.PIPE).communicate()[0]
        cpu_temp = p.decode("utf-8").strip()
    except Exception:
        cpu_temp = 0

    return cpu_temp`

![image](https://github.com/user-attachments/assets/d875bf7c-1f1d-4229-b559-12d33957c577)

**Comment by hjelev** [comment]
What is the error you get from the last version?
The code that works for you don't provides data after the floating point.
I prefer to fix the one I have so it also works for you.

**Reply by Fireblade900rr** [reply]
Is there any progress in this case ? I like to Monitor my CPU Temp in debian bookworm on a mini-pc
